src/layouts/default/AppBar.vue

```diff
@@ -23,4 +26,70 @@
 
 <script setup>
 import DefaultLogin from './Login.vue'
+import { useHdap } from '@/helpers/hdap'
+import { onMounted, ref } from 'vue'
+import router from '@/router'
+
+const hdap = useHdap()
+const namingContexts = ref([])
+const home = [{ title: 'Home', disabled: false, href: '/' }]
+const breadcrumbs = ref(home)
+
+onMounted(async () => {
+  if (namingContexts.value.length == 0) await setNamingContexts()
+  resetBreadcrumbs(namingContexts.value)
+})
+
+router.afterEach(() => {
+  resetBreadcrumbs(namingContexts.value)
+})
+
+async function setNamingContexts() {
+  const dns = await hdap.read('', { _fields: 'namingContexts' }, null)
```

Comment:
This should probably be factored out, since we might need to get the server capabilities elsewhere in the app. Fixed in my copy.